pkg/actors/internal/placement.go

```diff
@@ -129,70 +146,63 @@ func (p *ActorPlacement) AddHostedActorType(actorType string) error {
 
 // Start connects placement service to register to membership and send heartbeat
 // to report the current member status periodically.
-func (p *ActorPlacement) Start() {
-	p.serverIndex.Store(0)
-	p.shutdown.Store(false)
-
-	if established := p.establishStreamConn(); !established {
-		return
+func (p *ActorPlacement) Start(ctx context.Context) error {
+	if p.shutdown.Load() {
+		return errors.New("placement is shutdown")
 	}
 
-	// establish connection loop, whenever a disconnect occurs it starts to run trying to connect to a new server.
-	p.shutdownConnLoop.Add(1)
-	go func() {
-		defer p.shutdownConnLoop.Done()
-		for !p.shutdown.Load() {
-			// wait until disconnection occurs or shutdown is triggered
-			p.client.waitUntil(func(streamConnAlive bool) bool {
-				return !streamConnAlive || p.shutdown.Load()
-			})
-
-			if p.shutdown.Load() {
-				break
-			}
-			p.establishStreamConn()
+	for {
+		if err := p.manageConnectionLoop(ctx); err != nil {
+			return err
 		}
-	}()
-
-	// Establish receive channel to retrieve placement table update
-	p.shutdownConnLoop.Add(1)
-	go func() {
-		defer p.shutdownConnLoop.Done()
-		for !p.shutdown.Load() {
-			// Wait until stream is connected or shutdown is triggered.
-			p.client.waitUntil(func(streamAlive bool) bool {
-				return streamAlive || p.shutdown.Load()
-			})
-
-			resp, err := p.client.recv()
-			if p.shutdown.Load() {
-				break
-			}
 
-			// TODO: we may need to handle specific errors later.
-			if err != nil {
-				p.client.disconnectFn(func() {
-					p.onPlacementError(err) // depending on the returned error a new server could be used
-				})
-			} else {
-				p.onPlacementOrder(resp)
-			}
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-p.closedCh:
+			return nil
+		case <-p.clock.Tick(placementReconnectInterval):
```

Comment:
Please use an exponential backoff, which is reset when the connection is successful. See #6130